### PR TITLE
New device Fossil Hybrid and Skagen connected

### PR DESCRIPTION
**Pull request by dakhnod**
This adds support for Fossil hybrid and Skagen connected watches.

Work to do:
- remove Tasker dependency
- remove DeviceSupport [fix](https://github.com/dakhnod/Gadgetbridge/blob/09f6cada870122d2adc489021be34f88f9c34e6f/app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/DeviceCommunicationService.java#L820)
- add Button support for other models and Skagen watches

### CODE REVIEW

**Comment by dakhnod** [comment]
How can i retrieve the currently connected GBDevice from an activity opened through settings?

**Comment by ashimokawa** [comment]
@dakhnod 
Sorry, I did not have the time to review this yet.
Is the TODO in the PR description still valid?
Just let me know when you think you want this merged and I will start the review

**Comment by dakhnod** [comment]
@ashimokawa i was quite confident with my fork until fossil updated the firmware on the watches and broke everything.

Now i am unsure about what to do... i will try to implement a method of downgrading the watch...

**Comment by ashimokawa** [comment]
@dakhnod 
Woah, sorry, I was using the web gui to resolve the conflict in the README, and I did not know this will completely merge master into your branch. Hope I didn't break thinks on your side!

**Comment by dakhnod** [comment]
I regularilly try to merge the master branch myself, so no worries.

I will unWIP the Request whem it is ready to be reviewed...

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/TaskerPlugin.java [hunk 1]
```diff
@@ -0,0 +1,1001 @@
+package nodomain.freeyourgadget.gadgetbridge.devices.qhybrid;
```
Comment (cpfeiffer) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/TaskerPlugin.java]: This class looks like it is not really qhybrid specific, or is it? Where did you get it from? Who is the author and what's the license?

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/TimePicker.java [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.RadioButton;
+import android.widget.RadioGroup;
+import android.widget.ScrollView;
+
+import androidx.annotation.NonNull;
+import nodomain.freeyourgadget.gadgetbridge.service.devices.qhybrid.requests.PlayNotificationRequest;
+
+public class TimePicker extends AlertDialog.Builder {
```
Comment (cpfeiffer) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/TimePicker.java]: A whole custom TimePicker? Why not android.widget.TimePicker? Can you add a Javadoc comment about this if it is really needed?

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/externalevents/NotificationListener.java [hunk 3]
```diff
@@ -638,17 +653,14 @@ public void onNotificationRemoved(StatusBarNotification sbn) {
...
             }
         }
-        // FIXME: DISABLED for now
-        /*
         if (shouldIgnore(sbn))
             return;
 
         Prefs prefs = GBApplication.getPrefs();
-        if (prefs.getBoolean("autoremove_notifications", false)) {
+        if (prefs.getBoolean("autoremove_notifications", true)) {
```
Comment (cpfeiffer) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/externalevents/NotificationListener.java]: Maybe @ashimokawa has something to say about this default value change?

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/qhybrid/QHybridSupport.java [hunk 4]
```diff
@@ -0,0 +1,821 @@
...
+            try {
+                c.getSuperclass().getDeclaredMethod("handleResponse", BluetoothGattCharacteristic.class);
+                Request object = c.newInstance();
+                byte[] sequence = object.getStartSequence();
+                if (sequence.length > 1) {
+                    responseFilters.put((int) object.getStartSequence()[1], object);
+                    Log.d("Service", "response filter " + object.getStartSequence()[1] + ": " + c.getSimpleName());
+                }
+            } catch (NoSuchMethodException | IllegalAccessException | InstantiationException e) {
+                Log.d("Service", "skipping class " + c.getName());
```
Comment (cpfeiffer) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/qhybrid/QHybridSupport.java]: Please make these LOG.warn() instead, having such things in logfiles is really helpful at times.

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/qhybrid/QHybridSupport.java [hunk 5]
```diff
@@ -0,0 +1,821 @@
...
+                GBDeviceEventBatteryInfo batteryInfo = new GBDeviceEventBatteryInfo();
+                batteryInfo.level = gbDevice.getBatteryLevel();
+                batteryInfo.state = BatteryState.BATTERY_NORMAL;
+                handleGBDeviceEvent(batteryInfo);
+                break;
+            }
+            default: {
+                Log.d("Service", "unknown shit on " + characteristic.getUuid().toString() + ":  " + arrayToString(characteristic.getValue()));
+                try {
+                    File charLog = new File("/sdcard/qFiles/charLog.txt");
```
Comment (cpfeiffer) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/qhybrid/QHybridSupport.java]: Is this why we need WRITE_SDCARD permission now? Why not use Gadgebridge's data directory?

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/qhybrid/QHybridSupport.java [hunk 6]
```diff
@@ -0,0 +1,821 @@
...
+                    if (!charLog.exists()) {
+                        charLog.createNewFile();
+                    }
+
+                    FileOutputStream fos = new FileOutputStream(charLog, true);
```
Comment (cpfeiffer) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/qhybrid/QHybridSupport.java]: This stream is never closed, should be in a try-with-resources block.

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/qhybrid/QHybridSupport.java [hunk 7]
```diff
@@ -0,0 +1,821 @@
...
+        if (request instanceof GetStepGoalRequest) {
+            gbDevice.addDeviceInfo(new GenericItem(ITEM_STEP_GOAL, String.valueOf(((GetStepGoalRequest) request).stepGoal)));
+        } else if (request instanceof GetVibrationStrengthRequest) {
+            int strength = ((GetVibrationStrengthRequest) request).strength;
+            gbDevice.addDeviceInfo(new GenericItem(ITEM_VIBRATION_STRENGTH, String.valueOf(strength)));
+        } else if (request instanceof GetCurrentStepCountRequest) {
+            int steps = ((GetCurrentStepCountRequest) request).steps;
+            logger.debug("get current steps: " + steps);
+            try {
+                File f = new File("/sdcard/qFiles/");
```
Comment (cpfeiffer) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/qhybrid/QHybridSupport.java]: Here as well, why not in the private data directory?

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/qhybrid/QHybridSupport.java [hunk 8]
```diff
@@ -0,0 +1,821 @@
...
+                if (!f.exists()) f.mkdir();
+
+                File file = new File("/sdcard/qFiles/steps");
+                if (!file.exists()) {
+                    file.createNewFile();
+                }
+                logger.debug("Writing file " + file.getPath());
+                FileOutputStream fos = new FileOutputStream(file, true);
+                fos.write((System.currentTimeMillis() + ": " + steps + "\n").getBytes());
+                fos.close();
```
Comment (cpfeiffer) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/qhybrid/QHybridSupport.java]: try-with-resources and better log the exception.

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/qhybrid/QHybridSupport.java [hunk 9]
```diff
@@ -0,0 +1,821 @@
...
+                        e.printStackTrace();
+                        logger.debug("trash extra should be number 0.0-1.0");
+                    }
+                    break;
+                }
+            }
+        }
+    };
+
+    private final BroadcastReceiver commandReceiver = new BroadcastReceiver() {
```
Comment (cpfeiffer) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/qhybrid/QHybridSupport.java]: Why do you use broadcasts as communication channel instead of the `EventHandler` interface? This is supposed to be extended when new kinds of events/commands are needed.

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/DiscoveryActivity.java [hunk 10]
```diff
@@ -80,7 +80,7 @@
     private ScanCallback newLeScanCallback = null;
 
     // Disabled for testing, it seems worse for a few people
-    private final boolean disableNewBLEScanning = true;
+    private final boolean disableNewBLEScanning = false;
```
[the remark this answers is not in the document]
Comment (ashimokawa) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/DiscoveryActivity.java]: @cpfeiffer 
No this reverts a change of mine, people had problems with the new BLE scanning (which was only used on newer Android versions), I disabled it for all since I saw no reason.
But maybe I missed something

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/ConfigActivity.java [hunk 11]
```diff
@@ -0,0 +1,476 @@
...
+                et.setOnEditorActionListener(null);
+                final String text = device.getDeviceInfo(QHybridSupport.ITEM_STEP_GOAL).getDetails();
+                et.setText(text);
+                et.setSelection(text.length());
+                et.setOnEditorActionListener(new TextView.OnEditorActionListener() {
+                    @Override
+                    public boolean onEditorAction(TextView textView, int i, KeyEvent keyEvent) {
+                        if (i == EditorInfo.IME_ACTION_DONE || i == EditorInfo.IME_ACTION_NEXT) {
+                            String t = textView.getText().toString();
+                            if (!t.equals(null)) {
```
[the remark this answers is not in the document]
Comment (dakhnod) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/ConfigActivity.java]: don't know why i did this either...i will have a look at all of those suggestions when i feel somewhat done with the implementation...

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/DiscoveryActivity.java (hunk 10 above)
[the remark this answers is not in the document]
Comment (dakhnod) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/DiscoveryActivity.java]: Some people had issues finding the BLE devices i am implementing, so i enabled it since people using my fork mostly only use for those specific devices anyway...

Would it be a solution to implement a setting to enable the new scanning method?

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/ConfigActivity.java [hunk 12]
```diff
@@ -0,0 +1,476 @@
...
+                                LocalBroadcastManager.getInstance(ConfigActivity.this).sendBroadcast(intent);
+                                updateSettings();
+                            }
+                            ((InputMethodManager) getApplicationContext().getSystemService(Activity.INPUT_METHOD_SERVICE)).hideSoftInputFromWindow(getCurrentFocus().getWindowToken(), 0);
+                        }
+                        return true;
+                    }
+                });
+
+                if (device.getDeviceInfo(QHybridSupport.ITEM_EXTENDED_VIBRATION_SUPPORT).getDetails().equals("true")) {
```
[the remark this answers is not in the document]
Comment (dakhnod) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/ConfigActivity.java]: you are absolutely right, i have been trying to learn that for the last few years now...

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/ConfigActivity.java [hunk 13]
```diff
@@ -0,0 +1,476 @@
...
+    private void sendControl(PackageConfig config, String request) {
+        Intent intent = new Intent(request);
+        intent.putExtra("CONFIG", config);
+        LocalBroadcastManager.getInstance(this).sendBroadcast(intent);
+    }
+
+    private void refreshList() {
+        list.clear();
+        list.addAll(helper.getSettings());
+        list.add(null);
```
[the remark this answers is not in the document]
Comment (dakhnod) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/ConfigActivity.java]: the null-element symbolizes the add button at the end of the list, that is added by the ArrayAdapter

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/TaskerPlugin.java (hunk 1 above)
Comment (dakhnod) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/TaskerPlugin.java]: It is mentioned to download in [this](https://tasker.joaoapps.com/plugins-intro.html) tutorial, but i will remove anything tasker-related anyway in favor of intents

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/TimePicker.java (hunk 2 above)
Comment (dakhnod) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/TimePicker.java]: I wanted the user to be able to not select any hour/minute at all, and afaik the default picker doesn't allow that.

Also i wanted to have some specific setting in the same widget...

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/externalevents/NotificationListener.java (hunk 3 above)
Comment (dakhnod) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/externalevents/NotificationListener.java]: for now that was for testing anyway, i only changed that since it was not available in the settings to change...

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/qhybrid/QHybridSupport.java (hunk 5 above)
Comment (dakhnod) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/qhybrid/QHybridSupport.java]: I will address that.

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/DiscoveryActivity.java (hunk 10 above)
[the remark this answers is not in the document]
Comment (cpfeiffer) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/DiscoveryActivity.java]: I think we would start scanning with the method that would most probably work. That method may even depend on the Android version and/or the device capabilities. And then try the less likely methods.

A user probably has no idea which method to use. 

An alternative would be to first ask what kind of device to scan for and then decide on the method, but that would make it even less comfortable for the user.

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/ConfigActivity.java (hunk 11 above)
[the remark this answers is not in the document]
Comment (cpfeiffer) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/ConfigActivity.java]: :+1:

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/ConfigActivity.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (cpfeiffer) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/ConfigActivity.java]: From my experience this begs for `NullPointerExceptions` somewhere. But if that is indeed the best way to do it, I would add an explaining comment with a pointer to the place where the `null` value is handled.

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/PackageConfigHelper.java [hunk 14]
```diff
@@ -0,0 +1,124 @@
...
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.util.Log;
+
+import java.util.ArrayList;
+
+import nodomain.freeyourgadget.gadgetbridge.database.DBOpenHelper;
+import nodomain.freeyourgadget.gadgetbridge.service.devices.qhybrid.requests.PlayNotificationRequest;
+
+public class PackageConfigHelper extends DBOpenHelper {
```
[the remark this answers is not in the document]
Comment (cpfeiffer) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/PackageConfigHelper.java]: (If you need help with understanding/using/... greendao, just say so)

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/TaskerPlugin.java (hunk 1 above)
Comment (cpfeiffer) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/TaskerPlugin.java]: Alright.

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/TimePicker.java (hunk 2 above)
Comment (cpfeiffer) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/TimePicker.java]: OK, then I suggest adding this reasoning as a javadoc comment. If there's nothing fossil specific in there, I would also move it to a (new) fyg.gadgetbridge.ui package.

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/PackageConfigHelper.java (hunk 14 above)
[the remark this answers is not in the document]
Comment (dakhnod) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/PackageConfigHelper.java]: not that i know how to use greendao, i just didn't have the time to take a look at it yet...

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/DiscoveryActivity.java (hunk 10 above)
[the remark this answers is not in the document]
Comment (dakhnod) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/DiscoveryActivity.java]: I still think that simply leaving a checkbox somewhere to switch to the new method would be sufficient for now...

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/TimePicker.java (hunk 2 above)
Comment (dakhnod) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/TimePicker.java]: well, it is quite fossil specific, since only those devices  have the provided vibration patterns...

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/ConfigActivity.java [hunk 15]
```diff
@@ -0,0 +1,476 @@
...
+                layout2.addView(hourPicker);
+                TextView tw = new TextView(ConfigActivity.this);
+                tw.setText(":");
+                layout2.addView(tw);
+                layout2.addView(minPicker);
+
+                layout2.setGravity(Gravity.CENTER);
+
+                new AlertDialog.Builder(ConfigActivity.this)
+                        .setTitle("offset time by")
```
[the remark this answers is not in the document]
Comment (dakhnod) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/ConfigActivity.java]: i lazily extracted some resources, but due to the Codeberg translation that is not prio 1 right now...

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/ConfigActivity.java [hunk 16]
```diff
@@ -171,7 +219,8 @@ public void onFinish(boolean success, NotificationConfiguration config) {
                                                 helper.saveNotificationConfiguration(config);
                                                 LocalBroadcastManager.getInstance(ConfigActivity.this).sendBroadcast(new Intent(QHybridSupport.QHYBRID_COMMAND_NOTIFICATION_CONFIG_CHANGED));
                                             } catch (GBException e) {
-                                                e.printStackTrace();
+                                                GB.log("error saving configuration", GB.ERROR, e);
+
                                                 GB.toast("error saving notification", Toast.LENGTH_SHORT, GB.ERROR, e);
```
Comment (cpfeiffer) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/qhybrid/ConfigActivity.java]: I had tried to say that GB.toast() already logs error(!) messages, so no need to log *and* toast. Sorry for your time adding all those extra log statements :-}
Anyway, we can remove these on demand, they don't really hurt.

**Comment by dakhnod** [comment]
Is there anything critical left to do for the merge?
TBH i just want to get over with the Codeberg transition and focus on other development...

**Comment by ashimokawa** [comment]
@dakhnod 
I can try to merge this but I will at least remove all the apks, which landed here in your branch. Is that ok for you? apks definitely do not belong in a git repository.

Codeberg transition might take a bit longer since we have to clean up data from the closed branches (the jars from the other PR are still there even if you deleted your branch), as we wrote via mail, the repo grows from 16MB to 300MB after transition because of those.

Since this PR is the main showstopper I want to merge is ASAP, you can create new small PRs, but please keep your branches clean of binary files (jars and apks) from now on.

**Comment by dakhnod** [comment]
okay, sorry about the apks and jars.

File: app/build.gradle [hunk 17]
```diff
@@ -85,12 +85,21 @@ dependencies {
     implementation "net.e175.klaus:solarpositioning:0.0.9"
     // use pristine greendao instead of our custom version, since our custom jitpack-packaged
     // version contains way too much and our custom patches are in the generator only.
+
+    implementation 'com.twofortyfouram:android-plugin-client-sdk-for-locale:[4.0.3, 5.0['
+    implementation 'com.twofortyfouram:android-plugin-host-sdk-for-locale:[2.0.3,3.0['
+    implementation 'com.twofortyfouram:android-plugin-api-for-locale:[1.0.2,2.0['
```
Comment (ashimokawa) [inline app/build.gradle]: What is com.twofortyfouram:android-plugin-* ? Why is that needed?

File: app/build.gradle [hunk 18]
```diff
@@ -85,12 +85,21 @@ dependencies {
...
+
     implementation "org.greenrobot:greendao:2.2.1"
     implementation "org.apache.commons:commons-lang3:3.7"
     implementation "org.cyanogenmod:platform.sdk:6.0"
     implementation 'com.jaredrummler:colorpicker:1.0.2'
 //    implementation project(":DaoCore")
     implementation 'com.github.wax911:android-emojify:0.1.7'
+
+    implementation 'com.android.support:support-v4:28.1.0'
```
Comment (ashimokawa) [inline app/build.gradle]: Why is that needed?

File: app/build.gradle [hunk 19]
```diff
@@ -85,12 +85,21 @@ dependencies {
...
+
+    implementation 'com.android.support.constraint:constraint-layout:1.1.3'
```
Comment (ashimokawa) [inline app/build.gradle]: Why is that needed?

File: app/src/main/AndroidManifest.xml [hunk 20]
```diff
@@ -15,7 +15,7 @@
     <uses-permission android:name="android.permission.RECEIVE_SMS" />
     <uses-permission android:name="android.permission.SEND_SMS" />
     <uses-permission android:name="android.permission.READ_CONTACTS" />
-    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
+    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
```
Comment (ashimokawa) [inline app/src/main/AndroidManifest.xml]: Why is that needed?

File: app/build.gradle (hunk 17 above)
Comment (dakhnod) [reply app/build.gradle]: that has to do with tasker, i will try to remove that ASAP

File: app/build.gradle (hunk 18 above)
Comment (dakhnod) [reply app/build.gradle]: dunno RN, i will take a look at that

**Comment by ashimokawa** [comment]
initialization.txt does not belong in this PR I think?

File: app/build.gradle (hunk 18 above)
Comment (ashimokawa) [reply app/build.gradle]: We have androidx.legacy:legacy-support-v4:1.0.0 already as a dependency

File: app/build.gradle (hunk 19 above)
Comment (ashimokawa) [reply app/build.gradle]: We already have 
com.android.support.constraint:constraint-layout:1.1.3 in .... maybe just a duplicate, but not critical

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/DiscoveryActivity.java [hunk 21]
```diff
@@ -394,7 +394,11 @@ private void startDiscovery() {
             LOG.warn("Not starting discovery, because already scanning.");
             return;
         }
-        startDiscovery(Scanning.SCANNING_BT);
```
Comment (ashimokawa) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/DiscoveryActivity.java]: Why is notmal BT scanning disabled?

**Comment by ashimokawa** [comment]
Tell be when you are done with changes, I will try to clean up your branch afterwards (remove apk commits, minor other changes)

**Comment by ashimokawa** [comment]
I have removed all the apks from previous commits

**Comment by ashimokawa** [comment]
I have also removed the 200k initialization.txt file from this branch

**Comment by ashimokawa** [comment]
@dakhnod 
We merged this!

Probably a bit premature, tasker dependecies are still in, I missed these, I just thought there were still in build.gradle, but otherwise removed, my bad.

Can you do a new PR for that? Then we could release 0.40.0 with Fossil Q support soon! :+1:
